### PR TITLE
document going directly to display with ondiskgif

### PR DESCRIPTION
**Pull request by jepler**

### CODE REVIEW

File: shared-bindings/gifio/OnDiskGif.c [hunk 1]
```diff
@@ -69,6 +71,32 @@
...
+//|     ~6x depending on the GIF and display. This requires an LCD that uses
+//|     standard codes to set the update area, and which accepts RGB565_SWAPPED
+//|     pixel data directly:
+//|
+//|     .. code-block:: Python
+//|       # Initial set-up the same as above
+//|
+//|       # Take over display to drive directly
+//|       display.auto_refresh = False
+//|       bus = display.bus
```
Comment (gamblor21) [inline shared-bindings/gifio/OnDiskGif.c:86]: Is there a reason for this? display_bus is reference below

File: shared-bindings/gifio/OnDiskGif.c [hunk 2]
```diff
@@ -69,6 +71,32 @@
...
+//|           # Sleep for the frame delay specified by the GIF,
+//|           # minus the overhead measured to advance between frames.
+//|           time.sleep(max(0, next_delay - overhead))
+//|           next_delay = odg.next_frame()
+//|
+//|           display_bus.send(42, struct.pack(">hh", 0, odg.bitmap.width - 1))
+//|           display_bus.send(43, struct.pack(">hh", 0, odg.bitmap.height - 1))
+//|           display_bus.send(44, d.bitmap)
+//|
+//|           d.next_frame()
```
Comment (gamblor21) [inline shared-bindings/gifio/OnDiskGif.c:99]: You call `next_frame()` on line 93 too.

Comment (jepler) [reply shared-bindings/gifio/OnDiskGif.c]: thanks for both comments -- fixed now, I think?

**Review by gamblor21 — DISMISSED** [review]
Looks good to me. I've been using this to do some testing on my own so glad you shared it.

File: shared-bindings/gifio/OnDiskGif.c [hunk 3]
```diff
@@ -69,6 +71,30 @@
...
 //|           next_delay = odg.next_frame()
+//|
+//|     The displayio Group and TileGrid layers can be bypassed and the image can
+//|     be directly blitted to the full screen. This can give a speed-up of ~4x to
+//|     ~6x depending on the GIF and display. This requires an LCD that uses
+//|     standard codes to set the update area, and which accepts RGB565_SWAPPED
+//|     pixel data directly:
+//|
+//|     .. code-block:: Python
+//|       # Initial set-up the same as above
```
Comment (dhalbert) [inline shared-bindings/gifio/OnDiskGif.c:82]: Need a blank line to make sphinx happy, I think.
```suggestion
//|     .. code-block:: Python
//|
//|       # Initial set-up the same as above
```

**Review by dhalbert — CHANGES_REQUESTED** [review]

**Review by dhalbert — APPROVED** [review]
Thanks!

**Pull request merged**